### ISSUE TITLE
an implementation different from the Transformers

### ISSUE THREAD

**Issue by tangjiuqi097** [policy-gradient/GRPO-Zero#10]
Why has self.scaling been removed in the attention?

[transformers](https://github.com/huggingface/transformers/blob/32a58e31463e238c967207bf73772490c353551a/src/transformers/models/qwen2/modeling_qwen2.py#L241)

<img width="961" height="1282" alt="Image" src="https://github.com/user-attachments/assets/84789334-4cab-4bdd-94f4-33f691c9e548" />

But in [this repo](https://github.com/policy-gradient/GRPO-Zero/blob/d41bb486cfe4d58bf8dbb0b8f62b3c3102af9e74/qwen2_model.py#L139)

<img width="702" height="889" alt="Image" src="https://github.com/user-attachments/assets/9e88e343-de20-4b24-9cc4-733636831353" />

**Comment by NTT123** [comment]
Hi, the scaling factor `1/sqrt(head_dim)` is already taken into account by default in `torch.nn.functional.scaled_dot_product_attention`. See the documentation at https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html

**Comment by tangjiuqi097** [comment]
I got it. Thank you.
